lib/nrf_modem_lib/nrf91_sockets.c

```diff
@@ -752,9 +752,10 @@ static void nrf91_socket_offload_freeaddrinfo(struct zsock_addrinfo *root)
 	while (next != NULL) {
 		struct zsock_addrinfo *this = next;
 
-		next = next->ai_next;
 		k_free(this->ai_addr);
 		k_free(this);
+
+		next = next->ai_next;
```

Comment:
This is a bug, because `this = next` and thus `next` was freed just the line above.